### DISCUSSION TITLE
How far along is Fornjot?

### DISCUSSION THREAD

**Discussion by MRiabov** [hannobraun/fornjot#2457] [Q&A]
I'm an AI developer and I need a B-rep kernel. For the purpose of speed of my application, it better be GPU-native, but it can wait. 

I don't need sketches, but I need:

- CSG boolean subtraction, filets, changers and all the basic operations but w/o sketches,
- Solid drafts, 
- Export to STEP, 
- At least basic assembly - I don't need constraints, only an ability to place objects together. 

I personally started writing my own B-rep kernel for it, with useful tweaks for AI comprehension (all models also have an implicit version, which AI understands easily), but after 2 days I don't know if having B-rep on GPU would help that much. OpenCASCADE is simply buggy. 

So, would Fornjot be usable?

**Comment by hannobraun** [comment]
Hey @MRiabov,

I don't think you can use Fornjot. Literally none of the items you mention are currently implemented. At the speed things are going, it would take a long time to implement them.

I strongly recommend against doing your own b-rep kernel, if you plan to do _anything_ else. If there's an easy enough way to do it, I think it has yet to be discovered.

If you're looking for an open source kernel, I suspect that you can't do better than Open CASCADE. But you might want to give Truck a try: https://github.com/ricosjp/truck

Best of luck with your project!

**Reply by voxpelli** [reply]
And since @hannobraun is too humble to mention it himself I’m going to say it:

Also consider sponsoring @hannobraun to help him continue to work on this project so that it can reach all the goals that eg you need. Only that way will we arrive in a better state :) https://github.com/sponsors/hannobraun

(My company @yikesable is sponsoring)

**Reply by hannobraun** [reply]
Thank you, @voxpelli :smile:
